Switch to directory-based patch management

nathanchance opened the pull request:
This should make updating patches a little bit easier. I still envision the workflow being something like this:


1. Apply patches to Linux repo.
```
$ git am <patches_folder>/*.patch
```
2. Drop or modify patches.
3. Remove current patch set.
```
$ rm <patches_folder>/*.patch
```
4. Generate current patch set and update series file
```
$ git format-patch --base=<whatever> -o <patches_folder> <whatever>.. | sed "s;<patches_folder>/;;g" > <patches_folder>/series
```

Example on the last command:

```
$ git format-patch --base=v5.15-rc2 -o $CBL_GIT/ci/patches/mainline/ v5.15-rc2.. | sed "s;$CBL_GIT/ci/patches/mainline/;;g" > $CBL_GIT/ci/patches/mainline/series
```

I can document this later if it would be helpful. I am also open to exploring other models for managing patches.

nickdesaulniers commented on generate_workflow.py:
```diff
@@ -169,9 +170,11 @@ def get_steps(build, build_set):
+    patch_files = glob.glob(
+        ci_folder.joinpath(patches_folder).as_posix() + "/*.patch")
+    return " --patch-series {}".format(
+        patches_folder.as_posix()) if patch_files else ""
```
there's duplication between the additions to generate_workflow.py and generate_tuxsuite.py. Can you move this to utils.py and try to reuse more code between the two?

nickdesaulniers commented on check-patches.sh:
```diff
@@ -0,0 +1,51 @@
+        exit 1
+    fi
+
+    # Next, check that all of the patches in the series file exist (removed a patch, did not update series file)
```
is this comment < 80 lines?

nickdesaulniers commented on check-patches.sh:
```diff
@@ -0,0 +1,51 @@
+        fi
+    done <"${series}"
+
+    # Lastly, make sure that all of the patches in the patches folder are in the series file (removed from series file, did not remove patch file)
```
wrap comment?

nickdesaulniers reviewed (commented):
thanks for the patches! 😵‍💫 🦘

nickdesaulniers commented:
can you rebase on #202 to get the mulodi4 patch that I'm about to merge?

nathanchance commented:
Yes, I will address your comments and rebase so that I can drop the hack patch that I added (as mainline is not going to actually test your patch because we won't build the NBD code).

nathanchance commented:
Alright, I addressed all of your comments and rebased on #202, in addition to adding patches for most of the issues for trees based on -next or -tip. PTAL.

nickdesaulniers reviewed (approved):
looks great! nice work! 🟧 🟨

nathanchance commented:
Thanks for the review!
